iam/v1/service.proto

```diff
@@ -0,0 +1,168 @@
+}
+
+message ListNetworksResponse {
+  repeated org.packetbroker.v3.Network networks = 1;
+  // Total number of networks.
+  uint32 total = 2;
+}
+
+message CreateNetworkRequest {
+  org.packetbroker.v3.Network network = 1;
+}
+
+message NetworkRequest {
+  uint32 net_id = 1;
+}
+
+message ContactInfoValue {
+  org.packetbroker.v3.ContactInfo value = 1;
+}
+
+message UpdateNetworkRequest {
+  uint32 net_id = 1;
+  // Name of the tenant.
+  google.protobuf.StringValue name = 2;
+  // DevAddr blocks of the tenant.
+  DevAddrBlocksValue dev_addr_blocks = 3;
+  // Administrative contact.
+  ContactInfoValue administrative_contact = 4;
+  // Technical contact.
+  ContactInfoValue technical_contact = 5;
+}
+
+service NetworkRegistry {
+  // List networks.
+  rpc ListNetworks(ListNetworksRequest) returns (ListNetworksResponse);
+  // Create a network.
+  rpc CreateNetwork(CreateNetworkRequest) returns (org.packetbroker.v3.Network);
+  // Get a network.
+  rpc GetNetwork(NetworkRequest) returns (org.packetbroker.v3.Network);
+  // Update a network.
+  rpc UpdateNetwork(UpdateNetworkRequest) returns (google.protobuf.Empty);
+  // Delete a network.
+  rpc DeleteNetwork(NetworkRequest) returns (google.protobuf.Empty);
+}
+
+message ListTenantsRequest {
+  uint32 net_id = 1;
+  // Limit the number of items.
+  // If 0, use the server's default.
+  uint32 limit = 2;
+  // Number of items to skip.
+  uint32 offset = 3;
+}
+
+message ListTenantsResponse {
+  repeated org.packetbroker.v3.Tenant tenants = 1;
+  // Total number of tenants.
+  uint32 total = 2;
+}
+
+message CreateTenantRequest {
+  org.packetbroker.v3.Tenant tenant = 1;
+}
+
+message TenantRequest {
+  // LoRa Alliance NetID of the Member.
+  uint32 net_id = 1;
+  // ID assigned by the Member.
+  string tenant_id = 2;
+}
+
+message UpdateTenantRequest {
+  // LoRa Alliance NetID of the Member.
+  uint32 net_id = 1;
+  // ID assigned by the Member.
+  string tenant_id = 2;
+  // Name of the tenant.
+  google.protobuf.StringValue name = 3;
+  // DevAddr blocks of the tenant.
+  DevAddrBlocksValue dev_addr_blocks = 4;
+  // Administrative contact.
+  ContactInfoValue administrative_contact = 5;
+  // Technical contact.
+  ContactInfoValue technical_contact = 6;
+}
+
+service TenantRegistry {
+  // List a network's tenants.
+  rpc ListTenants(ListTenantsRequest) returns (ListTenantsResponse);
+  // Create a tenant.
+  rpc CreateTenant(CreateTenantRequest) returns (org.packetbroker.v3.Tenant);
+  // Get a tenant.
+  rpc GetTenant(TenantRequest) returns (org.packetbroker.v3.Tenant);
+  // Update a tenant.
+  rpc UpdateTenant(UpdateTenantRequest) returns (google.protobuf.Empty);
+  // Delete a tenant.
+  rpc DeleteTenant(TenantRequest) returns (google.protobuf.Empty);
+}
+
+message ListAPIKeysRequest {
+  // LoRa Alliance NetID of the Member.
+  uint32 net_id = 1;
+  // ID assigned by the Member (optional).
+  string tenant_id = 2;
+  // Cluster of the Member (optional).
+  string cluster_id = 3;
+}
```

Comment:
Shouldn't there be pagination here?

Reply:
There's currently no use case for many API keys, for now